core/templates/pages/contributor-dashboard-admin-page/topic-filter/topic-filter.component.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Topic filter component for the blog home page.
+ */
+
+import { Component, OnInit, ViewChild, ElementRef, Input, Output, EventEmitter} from '@angular/core';
+import { COMMA, ENTER } from '@angular/cdk/keycodes';
+import { MatAutocompleteTrigger } from '@angular/material/autocomplete';
+import { FormControl } from '@angular/forms';
+import { debounceTime, distinctUntilChanged, map, startWith } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'oppia-topic-filter',
+  templateUrl: './topic-filter.component.html'
+})
+
+export class TopicFilterComponent implements OnInit {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() listOfDefaultTopics!: string[];
+  @Input() selectedTopics: string[] = [];
```

Comment:
```suggestion
  @Input() defaultTopicNames!: string[];
  @Input() selectedTopicNames: string[] = [];
```
Nit: To differentiate between topic IDs and names.

Reply:
Done!